OpenRA.Mods.Common/Warheads/SpawnActorWarhead.cs

```diff
@@ -0,0 +1,226 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Warheads
+{
+	public enum SpawnFacing { Default, Projectile, Random }
+
+	[Desc("Spawns actors upon detonation.")]
+	public class SpawnActorWarhead : Warhead, IRulesetLoaded<WeaponInfo>
+	{
+		[FieldLoader.Require]
+		[ActorReference]
+		[Desc("Actors to spawn.")]
+		public readonly string[] Actors;
+
+		[Desc("The cell range to try placing the actors within.")]
+		public readonly int Range = 10;
+
+		[Desc("Map player to give the actors to. Defaults to the attacker's owner.")]
+		public readonly string Owner;
```

Comment:
This should use the same mechanism as SpawnActorOnDeath: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/SpawnActorOnDeath.cs#L35

Reply:
sorry, I tried, but that logic doesn't make sense as it's in reverse

Reply:
We don't need the "Victim" owner type, but it should be done similarly. If we are really against adding that (for just two options) then this should at least match the name "InternalOwner".